node/src/components/contract_runtime.rs

```diff
@@ -134,6 +151,33 @@ impl ExecutionPreState {
 
 type ExecQueue = Arc<Mutex<BTreeMap<u64, (FinalizedBlock, Vec<Deploy>, Vec<Deploy>)>>>;
 
+#[derive(Debug, From, Serialize)]
+pub(crate) enum Event {
+    // TODO: Consider boxing the internals of `ContractRuntimeRequest` to make the variant smaller
+    #[from]
+    ContractRuntimeRequest(Box<ContractRuntimeRequest>),
```

Comment:
The issue I have here is that the `ContractRuntimeRequest` (the type, not the variant) itself is large, it would probably be better to ensure that it is smaller by boxing some of its troublesome members. Otherwise a lot of newtype wrappers have to be boxed, which is not optimal.

Reply:
Size breakdown of the `ContractRuntimeRequest` type:
```
    variant `EnqueueBlockForExecution`: 407 bytes
    variant `CommitGenesis`: 31 bytes
    variant `Upgrade`: 31 bytes
    variant `Query`: 127 bytes
    variant `GetBalance`: 87 bytes
    variant `GetEraValidators`: 63 bytes
    variant `GetBids`: 55 bytes
    variant `IsBonded`: 271 bytes
    variant `GetTrie`: 63 bytes
    variant `GetTrieFull`: 55 bytes
    variant `PutTrie`: 31 bytes
    variant `FindMissingDescendantTrieKeys`: 55 bytes
    variant `ExecuteBlock`: 535 bytes
```

and Clippy complains because the ` TrieRequestIncoming(TrieRequestIncoming)` is only `88 bytes`.

I'll check if I can reduce the size of `ExecuteBlock` and `EnqueueBlockForExecution` (and maybe `IsBonded`) w/o too far-reaching refactoring.

Reply:
This commit shrinks the size of `FinalizedBlock` by boxing `proposer` and `era_report` fields: https://github.com/casper-network/casper-node/pull/2614/commits/161222fffeefa683ccbfd5e57e554d59cefdb734

It changes the `casper_types` crate, although the public interface stays intact.

Reply:
That's pretty good! What's the biggest remaining?

Reply:
Here are the current sizes in descending order:

```
variant `ExecuteBlock`: 279 bytes
variant `IsBonded`: 271 bytes
variant `EnqueueBlockForExecution`: 151 bytes
variant `Query`: 127 bytes
variant `GetBalance`: 87 bytes
variant `GetEraValidators`: 63 bytes
variant `GetTrie`: 63 bytes
variant `GetBids`: 55 bytes
variant `GetTrieFull`: 55 bytes
variant `FindMissingDescendantTrieKeys`: 55 bytes
variant `CommitGenesis`: 31 bytes
variant `Upgrade`: 31 bytes
variant `PutTrie`: 31 bytes
```

Reply:
So I guess the remaining culprits are `ExecuteBlock` and `IsBonded`? I'd assume for the former it may be enough to box the `FinalizedBlock`, which is already boxed in other instances. For `IsBonded`, the culprit the `PublicKey`.

Let's change these and see where we end up?

Reply:
Since these changes are not strictly necessary (yet helpful), I'll get back to them after finishing other tickets. If the PR is merged w/o additional changes I'll request a separate ticket dedicated to the size of `ContractRuntimeRequest`.